src/backend_types.mli

```diff
@@ -27,7 +36,11 @@ end
 module type Field_constrained = sig
   type 'field t
 
-  module Make (M : Field_prefix_intf) : S with type t = M.field t
+  module Make (F : Ctypes.FOREIGN) (M : Field_prefix_intf) :
+    S
+    with type t = M.field t
```

Comment:
maybe have F be a submodule inside of S so you can := both types at once with a module := ?

Reply:
@imeckler nice, I'll do this, thanks. Would be nice to do that as a separate PR though, I'd like to get this in so I can build get rid of the libffi stuff on Friday during the holiday!